### PR TITLE
Add query to get users group method and permit pagination

### LINKED ISSUE #443: The get_user_groups method of keycloak_admin doesn't allow paging, however in the Keycloak API it is available
Add a query dictionary so you can pass more optional parameters.

The method: https://github.com/marcospereirampj/python-keycloak/blob/e483bf4323c29e144448363bac23a157aa1d9328/src/keycloak/keycloak_admin.py#L870

Captura de Tela 2023-05-12 às 10 33 44

### PR DESCRIPTION
**Pull request by anapaulamendes**
Issue #443

### CODE REVIEW

File: src/keycloak/keycloak_admin.py [hunk 1]
```diff
@@ -867,24 +867,33 @@ def get_user(self, user_id):
...
         params_path = {"realm-name": self.connection.realm_name, "id": user_id}
-        data_raw = self.connection.raw_get(
-            urls_patterns.URL_ADMIN_USER_GROUPS.format(**params_path), **params
-        )
-        return raise_error_from_response(data_raw, KeycloakGetError)
+
+        url = urls_patterns.URL_ADMIN_USER_GROUPS.format(**params_path)
+
+        if "first" in query or "max" in query:
+            return self.__fetch_paginated(url, query)
```
Comment (ryshoooo) [inline src/keycloak/keycloak_admin.py:895]: Could you add a unit test that makes sure the pagination works as expected? :)

Comment (anapaulamendes) [reply src/keycloak/keycloak_admin.py:895]: Of course!

Comment (anapaulamendes) [reply src/keycloak/keycloak_admin.py:895]: Done!

Comment (anapaulamendes) [reply src/keycloak/keycloak_admin.py:895]: There is a pagination problem with "first" parameter only in keycloak version 20.0
I debugged the code to see where the error could be, but calling directly to keycloak 20.0 got the same problem. Can you help me @ryshoooo ?

Comment (anapaulamendes) [reply src/keycloak/keycloak_admin.py:895]: @ryshoooo  it seems to me that the problem is really in version 20.0. I found this issue listed in the Keycloak 21.0.0 released:

Release notes: https://www.keycloak.org/2023/02/keycloak-2100-released.html
Issue [#16899](https://github.com/keycloak/keycloak/issues/16899)

Comment (ryshoooo) [reply src/keycloak/keycloak_admin.py:895]: Right well, in some way, it's nice to know that our test framework catches Keycloak issues well :)

The assurance/claim we're giving to our users is that this package is meant to be working with the latest Keycloak version, so we don't necessarily have to be worried about older Keycloak versions for a new package release. The tests for previous Keycloak versions were also introduced in order to show that if the tests fail for the latest Keycloak but not for the older versions, the changes made in the PR are "OK", the tests are just not up-to-date with the latest Keycloak.

The correct way to resolve this, if we have tests depending on the Keycloak version to be >= X, is to have a decorator like

```python
@version(">=21")
def test_that_only_works_for_keycloak_21_or_above(...):
    ...
```

The decorator would check the Keycloak version and if the version does not satisfy the constraint, the test would be skipped.

It's not difficult to code such a decorator, but I'm also fine merging this with the tests failing and creating the decorator later :) @anapaulamendes Let me know whether you would like this to be merged or you want to have a stab at the decorator :)

Comment (anapaulamendes) [reply src/keycloak/keycloak_admin.py:895]: You can merge. Thanks @ryshoooo!

**Review by ryshoooo — APPROVED** [review]

**Pull request merged**